common/environment/environment.cpp

```diff
@@ -165,6 +245,30 @@ class CLocalEnvironment : implements IConstEnvironment, public CInterface
     unsigned getNumberOfDropZones() const { buildDropZoneCache(); return numOfDropZones; }
     IConstDropZoneInfo * getDropZoneByIndex(unsigned index) const;
     bool isDropZoneRestrictionEnabled() const;
+
+    virtual const char *getClusterKeyPairName(const char *cluster) const override
+    {
+        synchronized procedure(safeCache);
+        ensureClusterKeyMap();
```

Comment:
does it make sense to only call ensureclusterkeymap if !clusterKeyNameCache ?

Reply:
practically it makes no difference whether the condition is outside the function or inside (as it is).
A matter of style really, I prefer check inside and it's consistent with other methods in this file (i.e. buildMachineCache(), buildDropZoneCache)